File: backend/src/main/java/com/vehiclecfg/entities/Invoice.java
package com.vehiclecfg.entities;

import java.util.Date;

import jakarta.persistence.*;

@Entity
@Table(name = "invoice")
public class Invoice {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "inv_id")
    private Long invId;

    @Temporal(TemporalType.TIMESTAMP)
    @Column(name = "inv_date")
    private Date invDate;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "model_id", nullable = false)
    private Model model;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "id", nullable = false)
    private User user;

    @Column(name = "total_amt")
    private Double totalAmt;

    @Column(name = "tax")
    private Double tax;

    @Column(name = "net_amt")
    private Double netAmt;

    public Invoice() {
    }

    public Long getInvId() {
        return invId;
    }

    public void setInvId(Long invId) {
        this.invId = invId;
    }

    public Date getInvDate() {
        return invDate;
    }

    public void setInvDate(Date invDate) {
        this.invDate = invDate;
    }

    public Model getModel() {
        return model;
    }

    public void setModel(Model model) {
        this.model = model;
    }

    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }

    public Double getTotalAmt() {
        return totalAmt;
    }

    public void setTotalAmt(Double totalAmt) {
        this.totalAmt = totalAmt;
    }

    public Double getTax() {
        return tax;
    }

    public void setTax(Double tax) {
        this.tax = tax;
    }

    public Double getNetAmt() {
        return netAmt;
    }

    public void setNetAmt(Double netAmt) {
        this.netAmt = netAmt;
    }

}

File: backend-.NET/Vehiclecfg_group9/Vehiclecfg_group9/Models/Invoice.cs
using System;
using System.Collections.Generic;

namespace Vehiclecfg_group9.Models;

public partial class Invoice
{
    public long InvId { get; set; }

    public DateTime? InvDate { get; set; }

    public double? NetAmt { get; set; }

    public double? Tax { get; set; }

    public double? TotalAmt { get; set; }

    public int ModelId { get; set; }

    public int Id { get; set; }

    public virtual User IdNavigation { get; set; } = null!;

    public virtual ICollection<InvoiceDetail> InvoiceDetails { get; set; } = new List<InvoiceDetail>();

    public virtual ModelMaster Model { get; set; } = null!;
}
